client/alloc_runner.go

```diff
@@ -452,11 +482,24 @@ OUTER:
 	// Final state sync
 	r.syncStatus()
 
+	// Stop watching the shared allocation directory
+	r.ctx.AllocDir.StopDiskWatcher()
+
 	// Block until we should destroy the state of the alloc
 	r.handleDestroy()
 	r.logger.Printf("[DEBUG] client: terminating runner for alloc '%s'", r.alloc.ID)
 }
 
+// checkResources monitors and enforces alloc resource usage. It returns an
+// appropriate task event describing why the allocation had to be killed.
+func (r *AllocRunner) checkResources() (*structs.TaskEvent, string) {
+	if r.ctx.AllocDir.GetSize() > r.Alloc().Resources.DiskInBytes() {
+		return structs.NewTaskEvent(structs.TaskDiskExceeded),
```

Comment:
`NewTaskEvent(structs.TaskDiskExceeded).SetDiskLimit(limit).SetDiskSize(size)` that is what I meant would be nice to have